integrationTests/vm/arwen/erc20_test.go

```diff
@@ -0,0 +1,92 @@
+package arwen
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_SOL_002(t *testing.T) {
+	context := setupTestContext(t)
+
+	context.deploySC("./testdata/erc20sol002/0-0-2.wasm", "")
+
+	// Initial tokens and allowances
+	context.executeSC(&context.Owner, "transfer(address,uint256)@"+context.Alice.AddressHex()+"@"+formatHexNumber(1000))
+	context.executeSC(&context.Owner, "transfer(address,uint256)@"+context.Bob.AddressHex()+"@"+formatHexNumber(1000))
+
+	context.executeSC(&context.Alice, "increaseAllowance(address,uint256)@"+context.Bob.AddressHex()+"@"+formatHexNumber(500))
+	context.executeSC(&context.Alice, "decreaseAllowance(address,uint256)@"+context.Bob.AddressHex()+"@"+formatHexNumber(5))
+	context.executeSC(&context.Bob, "increaseAllowance(address,uint256)@"+context.Alice.AddressHex()+"@"+formatHexNumber(500))
+	context.executeSC(&context.Bob, "decreaseAllowance(address,uint256)@"+context.Alice.AddressHex()+"@"+formatHexNumber(5))
+
+	// Assertion
+	assert.Equal(t, uint64(42000), context.querySCInt("totalSupply()", [][]byte{}))
+	assert.Equal(t, uint64(1000), context.querySCInt("balanceOf(address)", [][]byte{context.Alice.Address}))
+	assert.Equal(t, uint64(1000), context.querySCInt("balanceOf(address)", [][]byte{context.Bob.Address}))
+	assert.Equal(t, uint64(495), context.querySCInt("allowance(address,address)", [][]byte{context.Alice.Address, context.Bob.Address}))
+	assert.Equal(t, uint64(495), context.querySCInt("allowance(address,address)", [][]byte{context.Bob.Address, context.Alice.Address}))
+
+	// Payments
+	context.executeSC(&context.Alice, "transferFrom(address,address,uint256)@"+context.Bob.AddressHex()+"@"+context.Carol.AddressHex()+"@"+formatHexNumber(50))
+	context.executeSC(&context.Bob, "transferFrom(address,address,uint256)@"+context.Alice.AddressHex()+"@"+context.Carol.AddressHex()+"@"+formatHexNumber(50))
+
+	// Assertion
+	assert.Equal(t, uint64(950), context.querySCInt("balanceOf(address)", [][]byte{context.Alice.Address}))
+	assert.Equal(t, uint64(950), context.querySCInt("balanceOf(address)", [][]byte{context.Bob.Address}))
+	assert.Equal(t, uint64(100), context.querySCInt("balanceOf(address)", [][]byte{context.Carol.Address}))
+}
+
+func Test_SOL_003(t *testing.T) {
+	context := setupTestContext(t)
+
+	context.deploySC("./testdata/erc20sol003/0-0-3.wasm", "")
+
+	// Minting
+	context.executeSC(&context.Owner, "transfer(address,uint256)@"+context.Alice.AddressHex()+"@"+formatHexNumber(1000))
```

Comment:
maybe you can use some constants

Reply:
Only if they increase test readability. Sometimes, additional entities (such as extra constants in this case) could clutter the test.